Training_Preparation.md

```diff
@@ -0,0 +1,403 @@
+PURIS backend internally handled the response. If we take a quick look at the logs of our PURIS backend containers, 
+we can see that the PartTypeInformationController has processed a request.
+
+### Task 4: Creating a New Sample Asset on Our Tractus-X EDC
+
+Now, we want to create a new sample asset the supplier's Tractus-X EDC in order to make it available for our customer partner.
+
+#### a) Asset Registration
+
+
+With this request, we create a new asset on the Tractus-X EDC of the supplier partner.
+
+Let's briefly look at the request body. We create a new fictitious API asset there, which we have given the identifier 
+"urn:samm:io.catenax.sample_information:1.0.0#SampleInformation".
+
+Since we do not have an endpoint in the PURIS backend that can serve this asset, we simply put a semi-randomly chosen 
+URL in the "baseUrl" field that responds with a list of artificial user names, when it gets called. 
+
+The other fields in the dataAddress object are steering the pull proxy settings. Here we allow to proxy the path but 
+nothing else (meaning we only allow GET operations). This information is basically defined by the Digital Twin KIT - 
+depending on your api endpoint, you may need to update different proxy properties individually. 
+
+
+#### b) Creating a Contract Definition
+
+Here, we reference the newly created asset in the request body.
+
+Additionally, we reference in the "accessPolicy" and "contractPolicy" fields two policies that our PURIS backend already 
+created on the Tractus-X EDC during its boot process. 
+
+If you're interested in further details in this regard, please see the [PURIS documentation](docs/architecture/08_concepts.md) 
+explaining the policies used and the data sovereignty concept. 
+
+### Task 5: Inserting the Sample Asset API into a Shell Descriptor
+
+In this step, we want to make our newly created asset available in a shell descriptor of the supplier's DTR. 
+
+#### a) Request a JWT token for DTR Authentication / Authorization
+You know the drill ;)
+
+#### b) Lookup for a specific Shell Descriptor
+As before.
+
+#### c) Querying a Specific Shell Descriptor
+
+If we briefly look at the response, we will of course notice that our SampleAsset API is not included there.
+
+#### d) Modifying an Existing Shell Descriptor
+
+Let's first calmly look at the exiting request body of this request. We have already manually inserted an entry 
+for our SampleAsset API there. With this PUT request, we will now ensure that it can now be found for the material.
+
+#### e) Querying a Specific Shell Descriptor Again
+
+Let's briefly check if the previous operation was successful.
+
+#### f) Task for Fast Completers:
+Redo the steps from Task 3 a to g. Do you also find our SampleAsset API?
+
+
+### Task 6: Testing Our Sample Asset API
+
+We are working from the customer's perspective here, and we will perform the necessary steps in order to obtain access 
+to the new Sample Asset, that the supplier has created recently. 
+
```

Comment:
We need to append some note here:

Note: This task is simplified as the sample endpoint does not provide the possibility to have `$value` suffix. Additionally, The DTR is not queried first - as a Fast Completer you did the query and likely found the API made available as a Submodel Descriptor. Also in that case you would normally need to extract the information for the href - same as previously - and append the `$value`.